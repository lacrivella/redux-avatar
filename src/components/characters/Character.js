import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  return (
    <div>
      <img src={character.img} />
      <p>{character.name}</p>
    </div>
  );
}

Character.propTypes = {
  character: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    img: PropTypes.string
  }).isRequired
};

export default Character;
