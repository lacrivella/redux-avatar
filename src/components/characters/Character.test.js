import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a App', () => {
    const wrapper = shallow(<Character character={{ id: 'hi', name: 'anng', img: 'url' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
