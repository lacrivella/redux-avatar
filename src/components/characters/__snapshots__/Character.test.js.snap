// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a App 1`] = `
<div>
  <img
    src="url"
  />
  <p>
    anng
  </p>
</div>
`;
